contracts/contracts/TACoApplication.sol

```diff
@@ -0,0 +1,708 @@
+        return stakingProviderInfo[_stakingProvider].authorized > 0;
+    }
+
+    /**
+    * @notice Returns true if operator has confirmed address
+    */
+    // TODO maybe _stakingProvider instead of _operator as input?
+    function isOperatorConfirmed(address _operator) public view returns (bool) {
+        address stakingProvider = _stakingProviderFromOperator[_operator];
+        StakingProviderInfo storage info = stakingProviderInfo[stakingProvider];
+        return info.operatorConfirmed;
+    }
+
+    /**
+    * @notice Return the length of the array of staking providers
+    */
+    function getStakingProvidersLength() external view returns (uint256) {
+        return stakingProviders.length;
+    }
+
+    /**
+    * @notice Bond operator
+    * @param _stakingProvider Staking provider address
+    * @param _operator Operator address. Must be a EOA, not a contract address
+    */
+    function bondOperator(address _stakingProvider, address _operator)
+        external onlyOwnerOrStakingProvider(_stakingProvider) updateReward(_stakingProvider)
+    {
+        StakingProviderInfo storage info = stakingProviderInfo[_stakingProvider];
+        address previousOperator = info.operator;
+        require(_operator != previousOperator, "Specified operator is already bonded with this provider");
+        // If this staker had a operator ...
+        if (previousOperator != address(0)) {
+            require(
+                !info.operatorConfirmed ||
+                block.timestamp >= uint256(info.operatorStartTimestamp) + minOperatorSeconds,
+                "Not enough time passed to change operator"
+            );
+            // Remove the old relation "operator->stakingProvider"
+            _stakingProviderFromOperator[previousOperator] = address(0);
+        }
+
+        if (_operator != address(0)) {
+            require(_stakingProviderFromOperator[_operator] == address(0), "Specified operator is already in use");
+            require(
+                _operator == _stakingProvider || getBeneficiary(_operator) == address(0),
+                "Specified operator is a provider"
+            );
+            // Set new operator->stakingProvider relation
+            _stakingProviderFromOperator[_operator] = _stakingProvider;
+        }
+
+        if (info.operatorStartTimestamp == 0) {
+            stakingProviders.push(_stakingProvider);
+        }
+
+        if (info.operatorConfirmed) {
+            authorizedOverall -= info.authorized;
+        }
+
+        // Bond new operator (or unbond if _operator == address(0))
+        info.operator = _operator;
+        info.operatorStartTimestamp = uint64(block.timestamp);
+        emit OperatorBonded(_stakingProvider, _operator, previousOperator, block.timestamp);
+        _releaseOperator(_stakingProvider);
+    }
+
+    /**
+    * @notice Make a confirmation by operator
+    */
+    function confirmOperatorAddress() external {
```

Comment:
Not for this PR, but I'd get rid of operator confirmation. See https://github.com/nucypher/nucypher-contracts/issues/103